GetDataType.js

```diff
@@ -1,11 +1,31 @@
-const dataTypes = [];
 function loadData(data) {
+  const dataTypes = [];
   const deff = data.library.statements.def;
   const filtered = deff.filter((d) => {
     return d.expression.operand !== undefined;
   });
+  function traverseELM(array) {
+    if (array.operand === undefined) {
+      for (let h = 0; h < array.length; h += 1) {
+        if (array[h].operand.dataType === undefined) {
+          traverseELM(array[h].operand);
+        } else dataTypes.push(array[h].operand.dataType);
+      }
+    }
+    return array.dataType;
+  }
+  let section;
   for (let i = 0; i < filtered.length; i += 1) {
-    dataTypes.push(filtered[i].expression.operand.dataType);
+    section = filtered[i].expression.operand;
+    if (section.dataType !== undefined) {
+      dataTypes.push(section.dataType);
+    } else {
+      section.forEach((o) => {
+        if (traverseELM(o.operand) !== undefined) {
+          dataTypes.push(traverseELM(o.operand));
```

Comment:
Another super small comment, but you are unnecessarily calling the traverse function twice here. Suggestion is to make a variable equal to the return result of `traverse(o.operand)` and then use that in both the if statement and the push call